geo/walk.py

```diff
@@ -1,31 +1,45 @@
 import networkx as nx
 import osmnx as ox
-from geopandas import GeoSeries
+from geopandas import GeoDataFrame
 from networkx import MultiDiGraph
-
+from shapely.geometry.point import Point
 
 
 '''Library to support geographical computations .'''
 
-def compute_route(start_point: tuple, end_point: tuple) -> list:
-  '''Returns the shortest path between a starting and end point 
 
-  Arguments:
-    start_point(tuple): a lat-lng point
-    end_point(tuple): a lat-lng point
-  Returns:
-    A list of points which construct the geometry of the path
-  
-  '''
+def compute_route(start_point: Point, end_point: Point, graph: MultiDiGraph, nodes: GeoDataFrame) -> list:
+    '''Returns the shortest path between a starting and end point 
+
+    Arguments:
+      start_point(Point): The lat-lng point of the origin point.
+      end_point(Point): The lat-lng point of the destination point.
+      graph(MultiDiGraph): The directed graph class that can store multiedges.
+      nodes(GeoDataFrame): The GeoDataFrame of graph nodes.
+    Returns:
+      A list of Points which construct the geometry of the path
+
+    '''
+
+    # get closest nodes to points
+    orig = ox.get_nearest_node(graph, tuple_from_point(start_point))
+    dest = ox.get_nearest_node(graph, tuple_from_point(end_point))
+
+    # get route
+    route = nx.shortest_path(graph, orig, dest, weight='length')  # shortest path
+    route_nodes = nodes[nodes['osmid'].isin(route)]
+    route_points=route_nodes['geometry']
+    return route_points.tolist()
+
+def tuple_from_point(point: Point):   
+    '''Returns the a lat-lng tuple
+
+    Arguments:
+      point(Point): A lat-lng point.
+    Returns:
+      A lat-lng tuple
 
-  G = ox.graph_from_place('Manhattan, New York City, New York, USA') #change to bounding box
-  nodes, streets = ox.graph_to_gdfs(G)
-  
-  #get closest nodes to points 
-  orig=ox.get_nearest_node(G,start_point)
-  dest=ox.get_nearest_node(G,end_point)
+    ''' 
 
+    return (point.y,point.x)
```

Comment:
Shouldn't this be the following?

`return (point.x, point.y)`

When you construct points elsewhere, the latitude comes first, e.g. `Point(FLAGS.orig_lat, FLAGS.orig_lon)`

Reply:
lon = x, lat = y

Reply:
But when you create the points, you give the latitude as the first argument, and Point.x() returns the first element:

https://github.com/Toblerity/Shapely/blob/24a9bf768315d7293bb31cc46d1f22a84cb8f7ea/shapely/geometry/point.py#L61

So how does lon=x?

BTW, the reviewer needs to resolve comments like this. :-)